game/serialization/src/main/kotlin/de/gleex/pltcmd/game/serialization/Storage.kt

```
@@ -0,0 +1,45 @@
+package de.gleex.pltcmd.game.serialization
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromByteArray
+import kotlinx.serialization.encodeToByteArray
+import kotlinx.serialization.protobuf.ProtoBuf
+import org.hexworks.cobalt.logging.api.LoggerFactory
+import java.io.File
+
+/**
+ * Manages persistent data.
+ */
+@OptIn(ExperimentalSerializationApi::class)
+internal object Storage {
+    private val log = LoggerFactory.getLogger(Storage::class)
+    private val dataFolder = File("data")
+
+    init {
+        if (!dataFolder.exists() && !dataFolder.mkdirs()) {
+            log.error("Failed to create folder ${dataFolder.absolutePath}")
```

And then...? Proceed with this error? Dead programs tell no lies. Either throw an exception and let it crash or make `Stoage` some kind of inactive so that you don't run into "some random IOException" when you try to save/load something.